### DISCUSSION TITLE
Displaying  authentication token data in enterprise audit log events - Beta Feedback

### DISCUSSION THREAD

**Discussion by boylejj** [community/community#35349] [Enterprise]
### Purpose
The purpose of this discussion is to collect feedback from Enterprises participating in the private beta for displaying token data used for authentication in enterprise audit log events. Enterprise owners interested in participating in the private beta should reach out to your GitHub account manager or [contact our sales team](https://github.com/enterprise/contact) to make the feature available for your enterprise.

**Note:** This feature is behind a feature flag that requires enabling before it can be used by an Enterprise. 

### Feature Summary
For enterprise audit log events where a token (PAT, OAuth, GitHub Apps) was  used as the method of authentication, the audit log will contain a hashed rendering of the token used for authentication, as well as the token type and token scope. In doing so, customers will be able to query actions a  token has been used to perform on enterprise assets. 

**Note:** Audit log exports will not have token data as part of private beta. 

### Intended Outcome
Compromised tokens have the potential to provide threat actors with access to private data. After learning of a token compromise, we know that Enterprise owners and security professionals want to understand the actions taken using the corrupt token. To support Enterprise customers to complete these investigations, we will enhance the audit log data by logging identifiable and queryable token data in the audit log for events using tokens for authentication.

### How will it work?
For additional information, please read about [identifying the token used for audit log events](https://docs.github.com/en/enterprise-cloud@latest/early-access/admin/articles/identifying-the-token-used-for-audit-log-events) in our early access documentation. Feedback on the documentation can also be provided by commenting on this discussion.

**Comment by amraljarhi** [comment]
Thank you for onboarding `algbra`. They are asking:

```
if the ability to pass SCIM data is on the product development roadmap? 
If not, is it possible I can make a feature request for it?

What I'm looking for is the SCIM username as part of the logs 
so that we can quickly and easily link logs from multiple sources 
to the same username. i.e., [user@example.com](mailto:user@example.com) 
will be the user's identity for 99% of systems but on GitHub the identity 
will be something completely different like java4life. Makes it a little harder 
to tell a story and threat hunt during an incident.
```

**Reply by boylejj** [reply]
This is something we're considering. We're planning to get SAML SSO metadata into the audit log in the very near future. I'll share the public roadmap link as soon as it is available.

**Reply by emtunc** [reply]
Would you be so kind as to email/message me when you have something? I'm more than happy to jump on a beta for this. Been waiting years!

**Reply by boylejj** [reply]
@emtunc Here is the changelog for the [Display SAML SSO authentication data in audit log – Private Beta](https://github.blog/changelog/2022-10-28-display-saml-sso-authentication-data-in-audit-log-private-beta/)

**Reply by emtunc** [reply]
> @emtunc Here is the changelog for the [Display SAML SSO authentication data in audit log – Private Beta](https://github.blog/changelog/2022-10-28-display-saml-sso-authentication-data-in-audit-log-private-beta/)

Amazing! @amraljarhi would you be so kind as to enrol our org into the program?

**Comment by cliffchapmanrbx** [comment]
Our team was dealing with a situation where a service account was configured with two separate tokens and we weren't sure which one was being used for which automation. The configured automations are in systems that don't allow us to reverse out the auth creds easily, so we left this as a mystery for a while.

By being able to review the logs for which token was involved with which changes we were better able to narrow it down and document it. This was very useful, thanks!

**Comment by jjjjjjjj987** [comment]
Would be nice to have token IDs on the audit log `personal_access_token` action category as well.

Currently, there is no way to know which token was requested (`personal_access_token.request_created`) or granted (`personal_access_token.access_granted`) on the audit logs.

We only have the token name shown at Web UI, making it harder for us to track their usage.
